WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -328,24 +366,11 @@ protected void onCancelled() {
         @Override
         protected SiteList doInBackground(Void... params) {
             List<Map<String, Object>> blogs;
-            String[] extraFields = {"isHidden", "dotcomFlag"};
 
-            if (mShowHiddenSites) {
-                if (mShowSelfHostedSites) {
-                    // all self-hosted blogs and all wp.com blogs
-                    blogs = WordPress.wpDB.getBlogsBy(null, extraFields);
-                } else {
-                    // only wp.com blogs
-                    blogs = WordPress.wpDB.getBlogsBy("dotcomFlag=1", extraFields);
-                }
+            if (mIsInSearchMode) {
+                blogs = WordPress.wpDB.getAllBlogs();
```

Comment:
Because the `extraFields` aren't being requested here, the "isHidden" field won't be returned - and this is necessary in order to provide [different styling for hidden sites](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2833-single-site-picker/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java#L157). Changing this so it uses `WordPress.wpDB.getBlogsBy(null, extraFields)` would resolve this.